compiler/test/dotty/tools/vulpix/ParallelTesting.scala

```diff
@@ -765,13 +754,36 @@ trait ParallelTesting extends RunnerOrchestration { self =>
           }
         }
 
+        def fail(msg: String): Unit = {
+          echo(msg)
+          failTestSource(testSource)
+        }
+
+        def reporterOutputLines(reporter: TestReporter): List[String] = {
+          reporter.allErrors.flatMap { error =>
+            (error.pos.span.toString + " in " + error.pos.source.file.name) :: error.getMessage().split("\n").toList
+          }
+        }
+        def checkFileTest(sourceName: String, checkFile: JFile, actual: List[String]) = {
+          if (checkFile.exists) {
+            val expexted = Source.fromFile(checkFile, "UTF-8").getLines().toList
+            diffMessage(sourceName, actual, expexted).foreach(fail)
+          }
+        }
```

Comment:
Minor: I usually find it more readable if we move the check `checkFile.exists` to the call site.

Reply:
done